### DISCUSSION TITLE
Unstable connection detected. Most often happening at 4 hours and 20 minutes

### DISCUSSION THREAD

**Discussion by henryjb99** [bubuntux/nordvpn#231] [Q&A] [answered]
I have been running the master version of this Docker image for a while, but lately, I have been getting a bunch of disconnects, eventually leading to the entire VPN dying and me having to restart my docker images. There doesn't seem to be a common factor in the amount of data transferred, but the connection never seems to last longer than 4 hours and 40 minutes. At some points, there can be bytes or kilobytes transfered in the 4+ hours, other times it could be MB or GB.

I am in the Midwest area of the USA using Comcast as my ISP and a wired connection to my router. My work VPN has not been having these issues, but they don't use Nord VPN either.

Has anyone else been experiencing these type of disconnects?

I have attached my logs showing the repeated disconnects.
[VPN_logs.txt](https://github.com/bubuntux/nordvpn/files/6911927/VPN_logs.txt)

Here are the times before the connection becomes unstable:
Uptime: 4 hours 20 minutes 26 seconds
Uptime: 35 minutes 21 seconds
Uptime: 1 hour 35 minutes 24 seconds
Uptime: 2 hours 10 minutes 25 seconds
Uptime: 5 minutes 20 seconds
Uptime: 40 minutes 22 seconds
Uptime: 3 hours 36 seconds
Uptime: 4 hours 20 minutes 36 seconds
Uptime: 4 hours 20 minutes 48 seconds
Uptime: 4 hours 20 minutes 50 seconds
Uptime: 4 hours 40 minutes 29 seconds
Uptime: 5 minutes
Uptime: 5 minutes
Uptime: 5 minutes
Uptime: 5 minutes
Uptime: 4 hours 20 minutes 27 seconds
Uptime: 4 hours 20 minutes 27 seconds

My docker-compose setup is:
services:
  vpn:
    image: ghcr.io/bubuntux/nordvpn:master
    container_name: vpn
    network_mode: bridge        # Required
    cap_add:
      - NET_ADMIN               # Required
      - SYS_MODULE              # Required for TECHNOLOGY=NordLynx
    sysctls:
      - net.ipv4.conf.all.rp_filter=2
    devices:
      - /dev/net/tun            # Required
    environment:                # Review https://github.com/bubuntux/nordvpn#environment-variables
      - USER=Super Secret Username     # Required
      - "PASS=Super Secret password"         # Required
      - CONNECT=P2P
      - TECHNOLOGY=NordLynx
      - NETWORK=Hidden
    ports:
      #Commented out
    restart: unless-stopped

**Comment by bubuntux** [comment, marked as answer]
yup, it works the same way for me, it's an issue on nordvpn site, the connection stalls after a while, so i put in place a mechanism to detect that and restart the connection, everything connected using the container may drop connection while the re-connection happens, but once it gets reestablished everything should work just fine.

**Reply by henryjb99** [reply]
Do you still run into the issue where the container just stops being able to reconnect after a while? For me, it is usually a day or so and then I have to bounce the entire pod.

**Reply by bubuntux** [reply]
It has being more stable for me on the master branch.
